### ISSUE TITLE
[arm/linux]Lots of tests failed in arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst

### ISSUE THREAD

**Issue by AriNuer** [dotnet/coreclr#21373]
**Job:**
https://ci.dot.net/job/dotnet_coreclr/job/master/view/arm/job/jitstress/job/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/39/

**Some failed logs:**
```
baseservices_exceptions._generics_nested_try_catch08_nested_try_catch08_._generics_nested_try_catch08_nested_try_catch08_sh [FAIL]
        /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/baseservices/exceptions/generics/nested-try-catch08/nested-try-catch08.sh: line 244: 15747 Illegal instruction     $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
        /n/nReturn code:      1/nRaw output file:      /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/baseservices/exceptions/Reports/baseservices.exceptions/generics/nested-try-catch08/nested-try-catch08.output.txt/nRaw output:/nBEGIN EXECUTION
        /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun nested-try-catch08.exe
        Test Passed
        Expected: 100
        Actual: 132
        END EXECUTION - FAILED
        Test Harness Exitcode is : 1
        /nTo run the test:/n> set CORE_ROOT=/ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/Tests/Core_Root/n> /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/baseservices/exceptions/generics/nested-try-catch08/nested-try-catch08.sh/n
        Expected: True
        Actual:   False
        Stack Trace:
          /mnt/j/workspace/dotnet_coreclr/master/arm_cross_checked_ubuntu/bin/tests/Linux.arm.Checked/TestWrappers/baseservices.exceptions/baseservices.exceptions.XUnitWrapper.cs(1439,0): at baseservices_exceptions._generics_nested_try_catch08_nested_try_catch08_._generics_nested_try_catch08_nested_try_catch08_sh()
```
```
JIT_Regression._VS_ia64_JIT_V1_2_M01_b10852_b10852_b10852_._VS_ia64_JIT_V1_2_M01_b10852_b10852_b10852_sh [FAIL]
        /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/JIT/Regression/VS-ia64-JIT/V1.2-M01/b10852/b10852/b10852.sh: line 244: 13538 Illegal instruction     $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
        /n/nReturn code:      1/nRaw output file:      /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/JIT/Regression/Reports/JIT.Regression/VS-ia64-JIT/V1.2-M01/b10852/b10852/b10852.output.txt/nRaw output:/nBEGIN EXECUTION
        /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun b10852.exe
        pass
        Expected: 100
        Actual: 132
        END EXECUTION - FAILED
        Test Harness Exitcode is : 1
        /nTo run the test:/n> set CORE_ROOT=/ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/Tests/Core_Root/n> /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/bin/tests/Linux.arm.Checked/JIT/Regression/VS-ia64-JIT/V1.2-M01/b10852/b10852/b10852.sh/n
        Expected: True
        Actual:   False
        Stack Trace:
          /mnt/j/workspace/dotnet_coreclr/master/arm_cross_checked_ubuntu/bin/tests/Linux.arm.Checked/TestWrappers/JIT.Regression/JIT.Regression.XUnitWrapper.cs(90339,0): at JIT_Regression._VS_ia64_JIT_V1_2_M01_b10852_b10852_b10852_._VS_ia64_JIT_V1_2_M01_b10852_b10852_b10852_sh()
```

**Comment by AndyAyersMS** [comment]
Will investigate.

**Comment by AndyAyersMS** [comment]
Exit code 132 is SIGILL, so perhaps related to #21582?

[CI history](https://ci.dot.net/job/dotnet_coreclr/job/master/view/arm/job/jitstress/job/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_jitstress2_tst/) doesn't show these exact tests failing repeatedly, but various other tests are, always with SIGILL. So would guess there's a race condition in GC stress that leads to intermittent failures.

**Comment by AndyAyersMS** [comment]
@janvorli is working up a fix for #21582 that should cover these failures too.

**Comment by AndyAyersMS** [comment]
Should be fixed by #23099.
